src/test/java/ui/tests/AuthTest.java

```diff
@@ -0,0 +1,193 @@
+        authenticationPage.logIn(user.getEmail(), user.getPassword());
+
+        logger.info("Open Personal Info page");
+        accountPage.openPersonalInfo();
+
+        logger.info("Create actual account");
+        UserAccountResult actualAccount =
+                new UserAccountResult(driver.findElement(By.name("firstname")).getAttribute("value"),
+                        driver.findElement(By.name("lastname")).getAttribute("value"),
+                        driver.findElement(By.name("email")).getAttribute("value"),
+                        new Select(driver.findElement(By.id("days"))).getFirstSelectedOption().getAttribute("value").trim(),
+                        new Select(driver.findElement(By.id("months"))).getFirstSelectedOption().getAttribute("value").trim(),
+                        new Select(driver.findElement(By.id("years"))).getFirstSelectedOption().getAttribute("value").trim());
+
+        logger.info("Verify created account");
+        Assert.assertEquals(expectedAccount, actualAccount);
+    }
+
+    @Test
+    public void testSuccessfulAccountUpdate() {
+        mainPage.clickSignInButton();
+        logger.info("Starting account registration with generated email");
+
+        logger.info("Test data generation for account update");
+        String updatedEmail = "FlynnUpdated+" + RandomStringUtils.randomAlphanumeric(10) + "@gmail.com";
+        String updatedFirstName = "JohnnyUpdated";
+        String updatedLastName = "FlynnUpdated";
+
+        UserAccountResult updatedAccount = new UserAccountResult(updatedFirstName, updatedLastName, updatedEmail,
+                "26", "10", "1985");
+
+        logger.info("Get User from User Account Pool");
+        userAccountsPool.deserialize("src/test/resources/accountData.json");
+        UserAccount user = userAccountsPool
+                .getUser("src/test/resources/accountData.json");
+
+        logger.info("Sign In");
+        mainPage.clickSignInButton();
+        authenticationPage.logIn(user.getEmail(), user.getPassword());
+        logger.info("Open Personal Info page");
+        accountPage.openPersonalInfo();
+
+        personalInformationPage.updateAccount(updatedAccount, user.getPassword());
+
+        Assert.assertTrue("Your personal information has been successfully updated.",
+                driver.findElement(By.xpath("//*[contains(@id, 'center_column')]")).getText()
+                        .contains("Your personal information has been successfully updated."));
+
+        logger.info("Signing out");
+        accountPage.signOut();
+        logger.info("Opening main page and clicking Sign in button");
+        mainPage.clickSignInButton();
+        logger.info("Sign In");
+        authenticationPage.logIn(updatedAccount.getEmail(), user.getPassword());
+        logger.info("Open Personal Info page");
+        accountPage.openPersonalInfo();
+
+        logger.info("Create actual account");
+        UserAccountResult actualAccount =
```

Comment:
That doesn't seem convenient. It looks like such code should be in some PageObject or smth like that.

Reply:
Agree, if you need to get some information from page, create a method in your page object to avoid `low-level` selenium interactions in test code